src/transformStateWithClones.js

```diff
@@ -7,7 +7,22 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  let newState = { ...state };
+  const states = [];
+
+  for (const key of actions) {
+    if (key.type === 'addProperties') {
+      Object.assign(newState, key.extraData);
```

Comment:
Ensure that `key.extraData` is an object when `key.type` is 'addProperties'. If `extraData` is not guaranteed to be an object, consider adding a validation step.